keps/sig-cluster-lifecycle/kubeadm/1739-customization-with-patches/README.md

```diff
@@ -0,0 +1,626 @@
+documentation such as release notes or a development roadmap.  It should be
+possible to collect this information before implementation begins in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself.  KEP editors, SIG Docs, and SIG PM
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+-->
+
+This KEP proposes a replacement for the feature introduced by the kubeadm KEP
+"20190722-Advanced-configurations-with-kubeadm-(Kustomize)".
+
+It has the same scope and goals, allowing the users to customize manifests
+generated by kubeadm using patches, but not using Kustomize as the backend
+for applying the patches.
+
+The existing Kustomize implementation in kubeadm is in Alpha state.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+The kubeadm team has decided that it is more beneficial if the project moves
+to using raw patches instead of Kustomize. Kustomize introduces an undesired
+dependency which can be avoided by using patches in a similar way kubectl
+uses them.
+
+### Goals
+
+<!--
+List the specific goals of the KEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+- Implement a solution for applying patches over kubeadm generated manifests
+during the common kubeadm commands "init", "join" and "upgrade" and their phases.
+- Deprecate and remove the existing Kustomize solution after a grace period
+of at least one release.
+- To initially allow patching of static Pod manifests only.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Validate if the user patches contain a good practice configuration and
+sane security values.
+- Allow patching of core addon configuration generated by kubeadm - e.g. CoreDNS,
+kube-proxy (until further notice).
+- Allowing patches to be added using the kubeadm ComponentConfig (until
+further notice).
+- Replace the kubeadm ComponentConfig. The sane defaults would still be
+generated by kubeadm.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+This KEP proposes introducing a new flag called `--experimental-customize`
```

Comment:
IMO we should drift away from customize or alias to avoid confusion. What about using patch/patches a a key world for the new flag?

Reply:
agreed, `--experimental-patches` SGTM. we can further agree during the kubeadm office hours on Wed.